v2/pkg/engine/resolve/loader.go

```diff
@@ -971,6 +973,30 @@ func (l *Loader) renderErrorsFailedToFetch(fetchItem *FetchItem, res *result, re
 	if err != nil {
 		return err
 	}
+
+	fmt.Printf("")
+
+	if l.apolloRouterCompatibilitySubrequestHTTPErrror && (res.statusCode < 200 || res.statusCode > 300) {
+		apolloRouterStatusErrorJSON := fmt.Sprintf(`{
+			"message": "HTTP fetch failed from '%[1]s': %[2]s",
+			"path": [],
+			"extensions": {
+				"code": "SUBREQUEST_HTTP_ERROR",
+				"service": "%[1]s",
+				"reason": "%[2]s",
+				"http": {
+					"status": %[3]d
+				}
+			}
+		}`, res.ds.Name, http.StatusText(res.statusCode), res.statusCode)
+		apolloRouterStatusError, err := astjson.ParseWithoutCache(apolloRouterStatusErrorJSON)
+		if err != nil {
+			return err
+		}
+
+		astjson.AppendToArray(l.resolvable.errors, apolloRouterStatusError)
+	}
+
 	l.setSubgraphStatusCode([]*astjson.Value{errorObject}, res.statusCode)
```

Comment:
could you reduce nesting by doing a negative check for the flag and returning early?
`if !l.apolloRouterCompatibilitySubrequestHTTPErrror || (res.statusCode >200 || res.statusCode < 300)`